src/verify/ResendButtonWithDelay.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import TextButton from 'src/components/TextButton'
+import colors from 'src/styles/colors'
+
+const RESEND_DELAY_TIME = 30 // recommended timeout for Twilio
+
+interface Props {
+  onPress(): void
+}
+
+function ResendButtonWithDelay({ onPress }: Props) {
+  const { t } = useTranslation()
+  const [secondsRemaining, setSecondsRemaining] = useState(RESEND_DELAY_TIME)
+
+  const handleOnPress = () => {
+    setSecondsRemaining(RESEND_DELAY_TIME)
+    onPress()
+  }
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setSecondsRemaining((prev) => (prev > 0 ? prev - 1 : 0))
```

Comment:
nit: it's a bit better to compute the difference from now and `startTime + RESEND_DELAY_TIME`. Timers can be delayed in case of user interactions or other activity happening. This way we ensure the the total time is closer to the real time value.

Reply:
yeah i suppose that's true, i feel like this is edge casey but maybe makes a difference for very old android devices? i updated and took some code from VerificationCountdown, i didn't share code since we'll delete VerificationCountdown soon